sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -96,8 +108,28 @@
               public Void apply(Iterable<Void> iter) {
                 return null; // ignore input
                 }
-            }))
-        .apply(Keys.<T>create());
+            }));
+    if (!acrossPanes) {
+      return combined.apply(Keys.<T>create());
```

Comment:
It is sort of boilerplate, but would make UIs a bit legible, to make separate composites for the two variants, like `StatefulDistinct` and `SinglePaneDistinct`.